### ISSUE TITLE
cURL Timeout

### ISSUE THREAD

**Issue by WebMedic-X** [osiset/Basic-Shopify-API#134]
Hi,

Is it normal to get a cURL timeout when accessing Shopify REST API?

How do you deal with it?

Also I noticed when the cURL timeout error happens, it doesn't fire the exception or returns any "errors" in the request, it just kills the process there.

For example this means it won't attempt to retry the request, like it would due to rate limiting or other HTTP status codes.

Thank you

**Comment by osiset** [comment]
Correct, there is no handling of timeout. We would need to catch the exception, determine if its a timeout, then run the retry process.
